### PR TITLE
vscode-extensions: add new extensions

### PR DESCRIPTION
**Pull request by TheMaxMur**
## Description of changes

1. **vscode-extensions.vscjava.vscode-java-pack**: init 0.25.2023121402
2. **vscode-extensions.visualstudioexptteam.vscodeintellicode**: init 1.2.30
3. **vscode-extensions.visualstudioexptteam.intellicode-api-usage-examples**: init 0.2.8
4. **vscode-extensions.ms-vscode.cpptools-extension-pack**: init 1.3.0
5. **vscode-extensions.stylelint.vscode-stylelint**: init 1.3.0
6. **vscode-extensions.reloadedextensions.reloaded-cpp**: init 0.1.9
7. **vscode-extensions.mathiasfrohlich.kotlin**: init 1.7.1
8. **vscode-extensions.hediet.vscode-drawio**: init 1.6.6
9. **vscode-extensions.hars.cppsnippets**: init 0.0.15
10. **vscode-extensions.firsttris.vscode-jest-runner**: init 0.4.72
11. **vscode-extensions.equinusocio.vsc-material-theme-icons**: init 3.5.0
12. **vscode-extensions.cweijan.dbclient-jdbc**: init 1.3.4
13. **vscode-extensions.13xforever.language-x86-64-assembly**: init 3.1.4

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
@ofborg build vscode-extensions.vscjava.vscode-java-pack vscode-extensions.visualstudioexptteam.vscodeintellicode vscode-extensions.visualstudioexptteam.intellicode-api-usage-examples vscode-extensions.ms-vscode.cpptools-extension-pack vscode-extensions.stylelint.vscode-stylelint vscode-extensions.reloadedextensions.reloaded-cpp vscode-extensions.mathiasfrohlich.kotlin vscode-extensions.hediet.vscode-drawio vscode-extensions.hars.cppsnippets vscode-extensions.firsttris.vscode-jest-runner vscode-extensions.equinusocio.vsc-material-theme-icons vscode-extensions.cweijan.dbclient-jdbc vscode-extensions.13xforever.language-x86-64-assembly

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -3280,6 +3408,22 @@ let
...
+          name = "reloaded-cpp";
+          publisher = "reloadedextensions";
+          version = "0.1.9";
+          sha256 = "sha256-KQiSD18W9NnsqhRt+XM3ko70u4zX4enn3OpMt0ebViU=";
+        };
+        meta = {
+          description = "C/C++ must-have highlighter that understands many coding styles and APIs. Use with 'Reloaded Themes' extension.";
+          downloadPage = "https://marketplace.visualstudio.com/items?itemName=reloadedextensions.reloaded-cpp";
+          homepage = "https://github.com/kobalicek/reloaded-cpp";
+          license = lib.licenses.unlicense;
```
Comment (jopejoe1) [inline pkgs/applications/editors/vscode/extensions/default.nix:3422]: This is licensed under the mit license and not the unlicense license.

```suggestion
          license = lib.licenses.mit;
```

**Review by jopejoe1 — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -2016,6 +2096,22 @@ let
...
+          name = "vscode-drawio";
+          publisher = "hediet";
+          version = "1.6.6";
+          sha256 = "sha256-SPcSnS7LnRL5gdiJIVsFaN7eccrUHSj9uQYIQZllm0M=";
+        };
+        meta = {
+          description = "This unofficial extension integrates Draw.io into VS Code.";
+          downloadPage = "https://marketplace.visualstudio.com/items?itemName=hediet.vscode-drawio";
+          homepage = "https://github.com/hediet/vscode-drawio";
+          license = lib.licenses.gpl3;
```
Comment (jopejoe1) [inline pkgs/applications/editors/vscode/extensions/default.nix:2110]: Missed this before, but gpl3 is deprecated and either gpl3Plus or gpl3Only should be used instead.

Comment (TheMaxMur) [reply pkgs/applications/editors/vscode/extensions/default.nix:2110]: are you sure? It is used in the project 
https://github.com/hediet/vscode-drawio/blob/master/LICENSE.md

Comment (TheMaxMur) [reply pkgs/applications/editors/vscode/extensions/default.nix]: @jopejoe1 Can you check the licenses now, are they all right?

**Comment by drupol** [comment]
@ofborg build vscode-extensions.vscjava.vscode-java-pack vscode-extensions.visualstudioexptteam.vscodeintellicode vscode-extensions.visualstudioexptteam.intellicode-api-usage-examples vscode-extensions.ms-vscode.cpptools-extension-pack vscode-extensions.stylelint.vscode-stylelint vscode-extensions.reloadedextensions.reloaded-cpp vscode-extensions.mathiasfrohlich.kotlin vscode-extensions.hediet.vscode-drawio vscode-extensions.hars.cppsnippets vscode-extensions.firsttris.vscode-jest-runner vscode-extensions.equinusocio.vsc-material-theme-icons vscode-extensions.cweijan.dbclient-jdbc vscode-extensions.13xforever.language-x86-64-assembly

**Review by jopejoe1 — APPROVED** [review]

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 2 above)
Comment (jopejoe1) [reply pkgs/applications/editors/vscode/extensions/default.nix]: Yeah, they look good now

**Comment by TheMaxMur** [comment]
@drupol It looks ready

**Comment by drupol** [comment]
Thanks!

**Pull request merged**